insights/parsers/dse_ldif.py

```diff
@@ -0,0 +1,240 @@
+===================================================
+
+This parser reads directory server's configuration
+file ``dse.ldif``.  The ``DSELdifConfig`` class
+implements the parsing and provides a ``list`` of all
+configuration lines present in the file.
+
+Sample configuration file::
+    # This is the RHDS configuration file. You should read the
+    # dse.ldif manual page in order to understand the options listed
+    #...
+dn:
+objectClass: top
+creatorsName: cn=server,cn=plugins,cn=config
+modifiersName: cn=server,cn=plugins,cn=config
+createTimestamp: 20151021124922Z
+modifyTimestamp: 20151021124922Z
+
+dn: cn=config
+cn: config
+objectClass: top
+..
+
+dn: cn=monitor
+objectClass: top
+objectClass: extensibleObject
+cn: monitor
+..
+
+dn: cn=encryption,cn=config
+objectClass: top
+objectClass: nsEncryptionConfig
+cn: encryption
+..
+
+dn: cn=features,cn=config
+objectClass: top
+objectClass: nsContainer
+cn: features
+numSubordinates: 5
+
+dn: cn=replication,cn=config
+objectClass: top
+objectClass: extensibleobject
+cn: replication
+..
+
+dn: cn=replSchema,cn=config
+objectClass: top
+objectClass: nsSchemaPolicy
+cn: replSchema
+..
+
+dn: cn=uniqueid generator,cn=config
+objectClass: top
+objectClass: extensibleObject
+nsState:: gK1YaqZe6gHTrx2O7V8kewAAAAAAAAAA
+cn: uniqueid generator
+..
+
+Examples:
+    >>> type(dse_ldif)
+    <class 'insights.parsers.dse_ldif.DSELdifConfig'>
+    >>> dse_ldif['creatorsName'] == ['cn=server,cn=plugins,cn=config']
+    True
+    >>> dse_ldif["numSubordinates"] == ['1']
+    True
+    >>> dse_ldif['modifyTimestamp'] == ['20151021124922Z']
+    True
+    >>> dse_ldif.last('numSubordinates') == '1'
+    True
+    >>> dse_ldif.get_values('nsCertfile') == ['alias/slapd-rhds10-2-cert8.db']
+    True
+    >>> dse_ldif.get_line('nsSSLClientAuth') == 'nsSSLClientAuth: allowed'
+    True
+    >>> next(iter(dse_ldif)).line == 'dn:'
+    True
+    >>> 'nsSSL3Ciphers' in dse_ldif
+    True
+    >>> dse_ldif['sslVersionMin'] == ['TLS1.0']
+    True
+    >>> dse_ldif.get('nsKeyfile')
+    [KeyValue(keyword='nsKeyfile', value='alias/slapd-rhds10-2-key3.db', kw_lower='nskeyfile', line='nsKeyfile: alias/slapd-rhds10-2-key3.db')]
+    >>> dse_ldif.line_uses_plus('nsTLS1')
+    True
+"""
+from collections import namedtuple
+from .. import Parser, parser, get_active_lines
+from insights.specs import Specs
+import re
+
+# Regex for matching 'Key:value' pairs seperated using :
+PLUS_PATTERN = re.compile(r'(.*)\:(.*)')
+
+
+@parser(Specs.dse_ldif)
+class DSELdifConfig(Parser):
+    """Parsing for ``/etc/dirsrv/slapd-<>/dse.ldif`` file.
+
+    Properties:
+        lines (list): List of `KeyValue` namedtupules for each line in
+            the configuration file.
+        keywords (set): Set of keywords present in the configuration
+            file, each keyword has been converted to lowercase.
+    """
+
+    KeyValue = namedtuple('KeyValue', ['keyword', 'value', 'kw_lower', 'line'])
```

Comment:
IMO, `kw_lower` can be replaced with `.lower()` where necessary.

Reply:
@jsvob Are these changes required all together?

Reply:
Not at all.

It's just a suggestion, that KeyValue might be simplified a bit by removing `kw_lower`, which is essentially a duplicate of `keyword.lower()`. That is, if the `kw_lower` attribute can be trivially computed from the `keyword` attribute, why have it at all?